ext/bin/pdb-dataset

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env bash
+
+set -ueo pipefail
+
+usage() { echo 'Usage: PDBBOX=BOX_DIR pdb-dataset import --timeshift-to now DUMPFILE'; }
+
+if test "$#" -lt 1 -o -z "$PDBBOX"; then
+  usage 1>&2
+  exit 1
+fi
+
+misuse()
+{
+  usage 1>&2
+  exit 1
+}
+
+timeshift=''
+timeshiftsql=''
+dumpfile=''
+command=''
+while test $# -gt 0; do
+  case "$1" in
+    --timeshift-to)
+      test $# -gt 0 || misuse
+      timeshift="$2"
+      shift 2
+      ;;
+    *)
+      if [ -z $command ]; then
+        command="$1"
+        shift
+      elif [ -z $dumpfile ]; then
+        dumpfile="$1"
+        shift
+      else
+        echo "Unknown argument: $1" 1>&2
+        misuse
+      fi
+  esac
+done
+
+tmpdir="$(mktemp -d "pdb-dataset-XXXXXX")"
+tmpdir="$(cd "$tmpdir" && pwd -P)"
+trap "$(printf 'rm -rf %q' "$tmpdir")" EXIT
+
+case "$command" in
+  import)
+    if [ "$timeshift" = 'now' ]; then
+      timshiftsql="now()"
+    elif [[ $timeshift =~ ^[0-9]{4}-[0-9]{2}-[0-9]{2}$ ]]; then
+      timshiftsql="TIMESTAMP '${timeshift}'"
+    else
+      echo "Invalid value for timeshift parameter: $timeshift" 1>&2
+      misuse
+    fi
+
+    echo "Checking for existing puppetdb database" 1>&2
+    db_intialized=$(psql "puppetdb" -t -c "SELECT 1 FROM information_schema.tables WHERE table_name = 'schema_migrations';")
+
+    if [ "$db_intialized" ]; then
+      echo 'puppetdb database already exists' 1>&2
+      exit 2
+    fi
+
+    echo "Restoring database from backup"
+    pgbox env pg_restore --role=postgres --no-owner --no-acl -d puppetdb "$dumpfile" || echo $?
```

Comment:
I suspect we don't need the `pgbox env` here (or anywhere else in the file), given the current requirement that we already be in a PDBBOX environment, since `pdbbox-env` already runs everything via `pgbox env`.  (See the last line in ext/bin/pdbbox-env`.)